### PR TITLE
[test] Add MPI Tool Information Interfaces (MPI_T) check

### PR DESCRIPTION
**Pull request by jgphpc**

### CODE REVIEW

**Comment by jgphpc** [comment]
kesch ci is in a weird state

**Comment by jgphpc** [comment]
@jenkins-cscs retest this

**Comment by jgphpc** [comment]
@jenkins-cscs retry all

File: cscs-checks/prgenv/mpi_t.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# Copyright 2016-2020 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
```
Comment (vkarak) [inline cscs-checks/prgenv/mpi_t.py:4]: Leave a new line here.

File: cscs-checks/prgenv/mpi_t.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        self.descr = 'Checks MPI_T control/performance variables/categories'
+        self.valid_systems = ['daint:gpu', 'dom:gpu', 'daint:mc', 'dom:mc']
+        self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-gnu', 'PrgEnv-pgi',
+                                    'PrgEnv-intel', 'PrgEnv-cray_classic']
+        self.build_system = 'SingleSource'
+        self.sourcepath = 'mpit_vars.c'
+        self.num_tasks_per_node = 1
+        self.variables = {'MPITEST_VERBOSE': '1', 'MPICH_VERSION_DISPLAY': '1'}
+        self.rpt = 'rpt'
+        self.executable_opts = ['&> %s' % self.rpt]
```
Comment (vkarak) [inline cscs-checks/prgenv/mpi_t.py:22]:
```suggestion
        self.executable_opts = [f'&> {self.rpt}']
```

File: cscs-checks/prgenv/mpi_t.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        rpt = os.path.join(self.stagedir, self.rpt)
+        self.run_cat_vars = sorted(sn.extractall(regex, rpt, 'category'))
+        # 4/ Extracted lists can be compared (when sorted):
+        self.sanity_patterns = sn.all([sn.assert_eq(self.ref_control_vars,
+                                                    self.run_control_vars),
+                                       sn.assert_eq(self.ref_perf_vars,
+                                                    self.run_perf_vars),
+                                       sn.assert_eq(self.ref_cat_vars,
+                                                    self.run_cat_vars),
+                                       ])
```
Comment (vkarak) [inline cscs-checks/prgenv/mpi_t.py:64]: Reformat as follows:

```suggestion
        self.sanity_patterns = sn.all([
            sn.assert_eq(self.ref_control_vars, self.run_control_vars),
            sn.assert_eq(self.ref_perf_vars, self.run_perf_vars),
            sn.assert_eq(self.ref_cat_vars, self.run_cat_vars),
        ])
```

File: cscs-checks/prgenv/mpi_t.py (hunk 1 above)
Comment (jgphpc) [reply cscs-checks/prgenv/mpi_t.py:4]: https://github.com/eth-cscs/reframe/pull/1251/commits/3d3aec801471a82105f54692b32e9bac7cfec273

**Pull request merged**